### PR TITLE
[MIG][15.0] base_rest: Migration to 15.0

### PR DESCRIPTION
**Pull request by yankinmax**

### CODE REVIEW

**Comment by yankinmax** [comment]
@simahawk @lmignon could you pls approve running workflow to proceed further?

File: oca_dependencies.txt [hunk 1]
```diff
@@ -0,0 +1,2 @@
+# See https://github.com/OCA/odoo-community.org/blob/master/website/Contribution/CONTRIBUTING.rst#oca_dependencies-txt
```
Comment (simahawk) [inline oca_dependencies.txt]: remove this

Comment (yankinmax) [reply oca_dependencies.txt]: forgot about this, sorry

File: base_rest/__manifest__.py [hunk 2]
```diff
@@ -6,18 +6,26 @@
...
-    "version": "14.0.4.3.1",
+    "version": "15.0.1.0.0",
     "development_status": "Beta",
     "license": "LGPL-3",
     "author": "ACSONE SA/NV, " "Odoo Community Association (OCA)",
     "maintainers": ["lmignon"],
     "website": "https://github.com/OCA/rest-framework",
     "depends": ["component"],
     "data": [
-        "views/assets_template.xml",
```
Comment (lmignon) [inline base_rest/__manifest__.py:17]: Why do you remove this file? Without the assets, the openapi UI will no more work.

Comment (yankinmax) [reply base_rest/__manifest__.py:17]: https://github.com/OCA/maintainer-tools/wiki/Migration-to-version-15.0
```
Assets are now directly declared in the manifest. You should move .js and .scss links from templates to __manifest__.py file.
```

Comment (yankinmax) [reply base_rest/__manifest__.py:17]: But the problem isn't with python code in checks. I can't understand what is wrong with README and index. Codecov check fails on that.

Comment (lmignon) [reply base_rest/__manifest__.py:17]: OK my bad. Thank you for the link.

Comment (yankinmax) [reply base_rest/__manifest__.py:17]: you're welcome

**Comment by yankinmax** [comment]
@lmignon can you pls help, how to make it always running all check tests?

**Comment by simahawk** [comment]
> @lmignon can you pls help, how to make it always running all check tests?

we cannot do anything about it AFAIK

**Comment by yankinmax** [comment]
so I should always ask to approve workflow?

**Comment by simahawk** [comment]
Reason https://docs.github.com/en/actions/managing-workflow-runs/approving-workflow-runs-from-public-forks

**Comment by yankinmax** [comment]
> Reason https://docs.github.com/en/actions/managing-workflow-runs/approving-workflow-runs-from-public-forks

thanks for the link

**Comment by yankinmax** [comment]
@simahawk so, from what I see now, there is no codecov message if "Files changed". I need to improve tests.

**Review by lmignon — APPROVED** [review]
Thank you for the migration. LGTM (Code review only)

File: base_rest/http.py [hunk 3]
```diff
@@ -2,7 +2,7 @@
 # Copyright 2017 Akretion (http://www.akretion.com).
 # @author Sébastien BEAU <sebastien.beau@akretion.com>
 # License LGPL-3.0 or later (http://www.gnu.org/licenses/lgpl).
-
+# flake8: noqa
```
Comment (simahawk) [inline base_rest/http.py]: reason for `noqa`?

**Review by simahawk — APPROVED** [review]

File: base_rest/http.py (hunk 3 above)
Comment (yankinmax) [reply base_rest/http.py]: pre-commit forces to "fix" here:
https://github.com/OCA/rest-framework/blob/feb7b9182bc5a4b5569e5f23411b259226276b48/base_rest/http.py#L121
Then if I add:
```
except ValueError as e:
      msg = "Invalid JSON data: %s" % str(e)
      _logger.info("%s: %s", self.httprequest.path, msg)
      raise BadRequest(msg) from e
```
It should be tested due to codecov, which seems not efficient for me. I tried actually, but couldn't succeed.
I tried to add `# flake8: disable=W1115` and it didn't work.
If this `noqa` isn't blocking I would leave it as it's.

Comment (simahawk) [reply base_rest/http.py]: but here you here you are disabling it for the whole file. IMO it's wrong.
IMO is wrong to turn of the coverage check at all, unless is something really trivial.
Please roll it back and ignore the coverage color.

**Review by simahawk — CHANGES_REQUESTED** [review]
rollback the noqa

**Review by simahawk — APPROVED** [review]

**Pull request merged**